src/util/domain_info_utils.c

```diff
@@ -197,7 +197,11 @@ find_domain_by_object_name(struct sss_domain_info *domain,
     }
 
     if (domainname == NULL) {
-        dom = domain;
+        if (strict) {
+            return NULL;
```

Comment:
Sumit, here you're leaking tmp_ctx.
Please, just do `dom = NULL` and let it normally follow the code path instead of directly returning here, so tmp_ctx is freed and NULL is returned.

Reply:
Good catch, fixed.